library/src/dotty/Show.scala

```diff
@@ -71,6 +71,10 @@ object Show {
     }) + "'"
   }
 
+  implicit val nothingShow: Show[Nothing] = new Show[Nothing] {
+    def show(x: Nothing) = ???
+  }
```

Comment:
Should we really need this? Does it show up for `showList`, have you tried to demand `implicit ev: T =:!= Nothing`?

Reply:
Currently:
```scala
scala> List() 
1 |List()
  |^
  |ambiguous implicit arguments of type dotty.Show[scala.collection.immutable.List[Nothing]] found for parameter ev of method show in class ShowValue.
  |I found:
  |
  |    dotty.Show.showList[T](/* ambiguous */implicitly[dotty.Show[T]])
  |
  |But both getter intShow in object Show and getter stringShow in object Show match type dotty.Show[T].
```
> have you tried to demand implicit ev: T =:!= Nothing? (in List / Map)

`=:!=` is not available by default. I quickly look it up online and it scared me. @OlivierBlanvillain This is your area of expertise. What do you think? Do you have an implementation of `=:!=` that works for Dotty? Is is preferable to an implicit `Show[Nothing]`?

Reply:
```scala
type =:!=[A, B] = implicits.Not[=:=[A, B]]
```

Reply:
Not sure which one is worse, that or a  `Show[Nothing]` :smile:

Reply:
Maybe our `Show` instances should not be so all-encompassing? :)

I mean, we need something equivalent to this mechanism for the REPL. But that doesn't mean we have to expose our users to it..